File: app/src/test/java/com/egecius/rxjava2_demo_2/rx/range/RangeExampleTest.java
package com.egecius.rxjava2_demo_2.rx.range;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.junit.MockitoJUnitRunner;

import io.reactivex.Flowable;
import io.reactivex.Observable;
import io.reactivex.observers.TestObserver;
import io.reactivex.subscribers.TestSubscriber;

@RunWith(MockitoJUnitRunner.class)
public class RangeExampleTest {

    private RangeExample mSut;

    @Before
    public void setUp() {
        mSut = new RangeExample();
    }

    @Test
    public void rangeObservable() {

        TestObserver<Integer> testObserver = Observable.range(5, 3).test();

        testObserver
                .assertComplete()
                .assertComplete()
                .assertValues(5, 6, 7);

    }

    @Test
    public void rangeFlowable() {

        TestSubscriber<Integer> testSubscriber = Flowable.range(5, 3).test();

        testSubscriber
                .assertComplete()
                .assertComplete()
                .assertValues(5, 6, 7);

    }

}

File: app/src/test/java/com/egecius/rxjava2_demo_2/rx/range/RangeExampleTest.kt
package com.egecius.rxjava2_demo_2.rx.range

import io.reactivex.Flowable
import io.reactivex.Observable
import org.junit.Test

class RangeExampleTest {

    @Test
    fun rangeObservable() {

        val testObserver = Observable.range(5, 3).test()

        testObserver
                .assertResult(5, 6, 7)
    }

    @Test
    fun rangeFlowable() {

        val testSubscriber = Flowable.range(5, 3).test()

        testSubscriber
                .assertResult(5, 6, 7)
    }

}